GitHubAction/GitHubAction.domain/Entities/Inputs.cs

```diff
@@ -0,0 +1,124 @@
+﻿
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+using Package.Domain.Enums;
+using Serilog;
+
+namespace GitHubAction.Domain.Entities;
+public class Inputs
+{
+    // always required
+    public string ApiKey { get; init; }
+    public TimeSpan TimeOut { get; init; }
+    public string Stage { get; init; }
```

Comment:
Stage can become an Enum, which will make things easier in other parts of the code.